### DISCUSSION TITLE
Deprecation cop for atom core

### DISCUSSION THREAD

**Discussion by oscar-defelice** [atom/atom#22196] [Q&A]
I updated atom to the last release, however I got (further than some package issues) this deprecation message. I am not able to solve it as if I try to open the `remote.js` file I got the error message, it does not exist. How can I fix this?

```bash 
Use require("electron").remote instead of require("remote")

Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/remote.js:4:6
Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/remote.js:11:3
Module._compile - <embedded>:11:149837
Object..js - <embedded>:11:153385
Module.load - internal/modules/cjs/loader.js:815:32
Module._load - internal/modules/cjs/loader.js:727:14
```

**Comment by xiaomingk** [comment]
I have the same problem. Now the whole Atom collapsed. It would be super nice if someone knows how to fix it. Many thanks!

**Reply by sadick254** [reply]
Which OS version and Atom version are we talking about?

**Reply by xiaomingk** [reply]
For me, it is Windows 10 and 1.56.0 the latest for Atom.

**Reply by sadick254** [reply]
Have you tried running atom in safe mode?

**Reply by oscar-defelice** [reply]
For me is OS X 11.0.1 and atom 1.56.0 (last version)

**Reply by xiaomingk** [reply]
> Have you tried running atom in safe mode?

The problem still exists in the safe mode!

**Reply by sadick254** [reply]
Unfortunately am not able to reproduce this. Could you send a screenshot and the reproduction steps. That would help.

**Reply by xiaomingk** [reply]
See the pic!
![image](https://user-images.githubusercontent.com/47573182/114764603-7202ab00-9d64-11eb-92f2-8a68bd98a478.png)

**Reply by icecream17** [reply]
Can you run in dev mode ```atom --dev``` and screenshot again? (The stacktraces are more helpful in dev mode)

**Comment by oscar-defelice** [comment]
Here is the screenshot ! thank you very much

![image](https://user-images.githubusercontent.com/49638680/114771205-9498c200-9d6c-11eb-9ff0-f773a4a3f52e.png)

when I actually try to open the files raising the issues, I got an error (they do not exist).

**Reply by tomasrose** [reply]
I'm having a very similar problem. But apart from the deprecation that Oscar is showing, I'm also having another one:
Use require("electron").shell instead of require("shell")

Object.<anonymous> - C:\Users\Tomás\AppData\Local\atom\app-1.56.0\resources\app.asar\exports\shell.js:4:6
Object.<anonymous> - C:\Users\Tomás\AppData\Local\atom\app-1.56.0\resources\app.asar\exports\shell.js:9:3
Module._compile - <embedded>:11:149837
Object..js - <embedded>:11:153385
Module.load - internal/modules/cjs/loader.js:815:32
Module._load - internal/modules/cjs/loader.js:727:14

anyone have an idea of what's happening? My Atom stopped working completely

**Reply by sadick254** [reply]
@oscar-defelice @xiaomingk Could you try deleting the `.atom` folder. Uninstalling atom and reinstalling again.

**Reply by UziTech** [reply]
Probably back up your settings (`.atom/config.cson`) first

**Reply by oscar-defelice** [reply]
> @oscar-defelice @xiaomingk Could you try deleting the `.atom` folder. Uninstalling atom and reinstalling again. 

@sadick254 already done, I completely deleted the .atom folder and reinstalled atom. It didn't work

**Comment by zubairsediqi** [comment]
After the last update, the Atom does not display pdf and I get this issue: 

-----
atom core (2 deprecations)
Use require("electron").shell instead of require("shell")

Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/shell.js:4:6
Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/shell.js:9:3
Module._compile - <embedded>:11:149837
Object..js - <embedded>:11:153385
Module.load - internal/modules/cjs/loader.js:815:32
Module._load - internal/modules/cjs/loader.js:727:14
Use require("electron").remote instead of require("remote")

Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/remote.js:4:6
Object.<anonymous> - /Applications/Atom.app/Contents/Resources/app.asar/exports/remote.js:11:3
Module._compile - <embedded>:11:149837
Object..js - <embedded>:11:153385
Module.load - internal/modules/cjs/loader.js:815:32
Module._load - internal/modules/cjs/loader.js:727:14
-------------

I run the atom on MAC OS Big Sur 11.1   and the atom version is: 1.56.0, removed and reinstalled the atom, but no luck. 

Does anyone know how to fix it?

**Reply by tomasrose** [reply]
Hey guys, 
yesterday, someone suggested me trying the following:

First, close all your Atom instances and then open your cmd (I have windows) and write the following:

apm uninstall julia-client ink
apm install julia-client ink


after uninstalling and re-installing these,  atom is working again (for now at least). The two deprecations are still showing, but it seems that those were not the real issue 

http://docs.junolab.org/latest/man/update/

(i need to keep these instructions for the next atom update)

**Reply by zubairsediqi** [reply]
@tomasrose 
Thanks for your sugesstion. 
Tried uninstalling and re-installing julia-client in my MAC, no effect. 
BTW - in my atom, I have not had  "julia-client ink" so basically I installed a fresh copy.

**Comment by max123kl** [comment]
See my comment here:
https://github.com/atom/atom/issues/22208#issuecomment-821866586

**Comment by oscar-defelice** [comment]
For the moment, only for those it was only a visual issue: I make it disappear by going into `style.less` and adding the following lines

```less
// make deprecation-cop warning invisible
.status-bar .deprecation-cop-status {
  display: none;
}
```

**Comment by krcabrer** [comment]
I got the same problem but in Ubuntu 20.04.
Any update besides just deactivated the Deprecation Cop?
Thank you for your help.

**Comment by tp1050** [comment]
SAME HERE!!

`atom core (1 deprecation)
Use require("electron").shell instead of require("shell")

Object.<anonymous> - /usr/share/atom/resources/app.asar/exports/shell.js:4:6
Object.<anonymous> - /usr/share/atom/resources/app.asar/exports/shell.js:9:3
Module._compile - <embedded>:11:149891
Object..js - <embedded>:11:153485
Module.load - internal/modules/cjs/loader.js:815:32
Module._load - internal/modules/cjs/loader.js:727:14`

**Comment by icecream17** [comment]
Duplicate of https://github.com/atom/atom/issues/22306
